x/cronos/keeper/precompiles/relayer.go

```diff
@@ -191,8 +199,34 @@ func (bc *RelayerContract) Run(evm *vm.EVM, contract *vm.Contract, readonly bool
 	if err != nil {
 		return nil, errors.New("fail to unpack input arguments")
 	}
-	input := args[0].([]byte)
 	converter := cronosevents.RelayerConvertEvent
+	if method.Name == RegisterPayee || method.Name == RegisterCounterpartyPayee {
+		execErr := stateDB.ExecuteNativeAction(precompileAddr, converter, func(ctx sdk.Context) error {
+			portID := args[0].(string)
+			channelID := args[1].(string)
+			caller := sdk.AccAddress(contract.CallerAddress.Bytes()).String()
+			if method.Name == RegisterPayee {
+				payeeAddr := sdk.AccAddress(args[2].(common.Address).Bytes()).String()
+				_, err := bc.ibcFeeKeeper.RegisterPayee(
+					ctx,
+					ibcfeetypes.NewMsgRegisterPayee(portID, channelID, caller, payeeAddr),
```

Comment:
what about handle it similarly to the other relayer msgs, would that be simpler?